linux/src/srw_lock_linux.c

```diff
@@ -0,0 +1,144 @@
+// Copyright (C) Microsoft Corporation. All rights reserved.
+
+#include <stdlib.h>              // for NULL, free, malloc
+#include <stdbool.h>
+
+#include <pthread.h>
+
+#include "c_logging/xlogging.h"
+
+#include "c_pal/srw_lock.h"
+
+typedef struct SRW_LOCK_HANDLE_DATA_TAG
+{
+    pthread_rwlock_t rwlock;
+} SRW_LOCK_HANDLE_DATA;
+
+SRW_LOCK_HANDLE srw_lock_create(bool do_statistics, const char* lock_name)
+{
+    (void)do_statistics;
+    (void)lock_name;
+    SRW_LOCK_HANDLE result;
+
+    result = malloc(sizeof(SRW_LOCK_HANDLE_DATA));
+    if (result == NULL)
+    {
+        /*return as is*/
+        LogError("failure in malloc(sizeof(SRW_LOCK_HANDLE_DATA)=%zu)", sizeof(SRW_LOCK_HANDLE_DATA));
+    }
+    else
+    {
+        pthread_rwlock_init(&result->rwlock, NULL);
```

Comment:
>pthread_rwlock_init [](http://example.com/codeflow?start=8&length=19)

This function returns something per [https://pubs.opengroup.org/onlinepubs/7908799/xsh/pthread_rwlock_init.html](https://pubs.opengroup.org/onlinepubs/7908799/xsh/pthread_rwlock_init.html)